dm/master/openapi.go

```diff
@@ -215,12 +222,72 @@ func (s *Server) DMAPIGetSourceStatus(ctx echo.Context, sourceName string) error
 
 // DMAPIStartTask url is:(POST /api/v1/tasks).
 func (s *Server) DMAPIStartTask(ctx echo.Context) error {
-	return nil
+	var req openapi.CreateTaskRequest
+	if err := ctx.Bind(&req); err != nil {
+		return err
+	}
+	task := req.Task
+	if err := adjustModelTask(&task); err != nil {
```

Comment:
seems we can reuse `Server.generateSubTask`

Reply:
It doesn't work, because `Server.generateSubTask` converts the task yaml to a subtask, but there is no yaml here

Reply:
I'm not sure if it's easier that we first turn API request to TaskConfig and reuse them

Reply:
and slightly recommend that implement `(*Task).Adjust` in a separate file under `openapi` package

Reply:
Maybe it would be better to go straight to `subtaskcfg`? Because this is the configuration that is actually sent to the worker, and also `subtaskcfg` also can be reversed to `TaskConfig`

Reply:
I noticed that lots of routines can be shared between Task -> []SubTaskConfig and TaskConfig -> []SubTaskConfig

if in future we need to add a field in TaskConfig and Task, we'd better have an unified place to copy that field to SubTaskConfig

Reply:
SGTM, but this PR i already too large, i will do this job in next PR(task status/list API)

Reply:
wii do this job in https://github.com/pingcap/dm/pull/2169